### DISCUSSION TITLE
Building JWebAssembly v0.4 failing lots of tests

### DISCUSSION THREAD

**Discussion by zlgonzalez** [i-net-software/JWebAssembly#61] [General]
Hi,
  I'm on Ubuntu 22.04. I am building JWebAssembly v0.4, but it fails with the following error:

1205 tests completed, 963 failed, 16 skipped

  Did I miss installing anything?

Thanks,
Ron

**Comment by Horcrux7** [comment]
Currently it works only with Java 8. Java 11 is in progress.

**Reply by zlgonzalez** [reply]
I tried Temurin JDK 8, and it fails with a class file issue incompatibility, despite deleting $HOME/.gradle and building from scratch.

Thoughts?

Thanks,
Ron

(base) rongonzalez@pop-os:~/Documents/code/JWebAssembly$ gradle clean build

Welcome to Gradle 7.5.1!

Here are the highlights of this release:
 - Support for Java 18
 - Support for building with Groovy 4
 - Much more responsive continuous builds
 - Improved diagnostics for dependency resolution

For more details see https://docs.gradle.org/7.5.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

> Task :javadoc
/home/rongonzalez/Documents/code/JWebAssembly/src/de/inetsoftware/jwebassembly/module/nativecode/ReplacementForEnums.java:34: warning: no @return
    public native String name();
                         ^
/home/rongonzalez/Documents/code/JWebAssembly/src/de/inetsoftware/jwebassembly/module/nativecode/ReplacementForEnums.java:39: warning: no @return
    public native static ReplacementForEnums[] values();
                                               ^
2 warnings

> Task :compileTestJava
/home/rongonzalez/Documents/code/JWebAssembly/test/de/inetsoftware/jwebassembly/module/StructTypeTest.java:21: error: cannot access org.mockito.Mockito
import static org.mockito.Mockito.mock;
                         ^
  bad class file: /home/rongonzalez/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/5.3.1/7cac313592a29ae5e29c52c22b15c3ae5ab561b2/mockito-core-5.3.1.jar(org/mockito/Mockito.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.

> Task :compileTestJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 30s
8 actionable tasks: 8 executed

**Comment by Horcrux7** [comment]
Do you try to compile the old release snapshot 0.4 or any other old stand? 

The current [build script](https://github.com/i-net-software/JWebAssembly/blob/master/build.gradle) use mockito-core 4.x because the problems with the new version 5.x

**Reply by zlgonzalez** [reply]
I cloned the v0.4 git branch. Looks like I should go to main?

**Reply by zlgonzalez** [reply]
Yeah that fixed it - thanks. That said, I would prefer using a released version. Will current main branch be released as v1.0 at some point?

**Comment by Horcrux7** [comment]
Yes, it should be released as 1.0. Before I will see to fix the Java 11 problems.
